File: idscp2/src/main/java/de/fhg/aisec/ids/idscp2/idscp_core/Idscp2MessageListener.java
package de.fhg.aisec.ids.idscp2.idscp_core;

/**
 * An interface for an IDSCP message listener
 */
public interface Idscp2MessageListener {

    /*
     * notify the listener about new data
     */
    void onMessage(String type, byte[] data);
}

File: idscp2/src/main/kotlin/de/fhg/aisec/ids/idscp2/idscp_core/api/idscp_connection/Idscp2MessageListener.kt
package de.fhg.aisec.ids.idscp2.idscp_core.api.idscp_connection

import de.fhg.aisec.ids.idscp2.idscp_core.api.idscp_connection.Idscp2Connection

/**
 * An interface for an IDSCP message listener
 */
fun interface Idscp2MessageListener {
    /*
     * notify the listener about new data
     */
    fun onMessage(connection: Idscp2Connection, data: ByteArray)
}